cf/s3-v2.yaml

```diff
@@ -0,0 +1,265 @@
+AWSTemplateFormatVersion: "2010-09-09"
+Description: Set up S3 buckets for Cellenics [managed by github.com/hms-dbmi-cellenics/iac]
+
+Parameters:
+  Environment:
+    Type: String
+    Default: development
+    AllowedValues:
+      - development
+      - staging
+      - production
+    Description: The environment for which the buckets need to be created.
+
+Resources:
+  PlotsTablesBucket:
+    Type: AWS::S3::Bucket
+    Properties:
+      BucketName: !Sub "plots-tables-${Environment}-${AWS::AccountId}"
+      PublicAccessBlockConfiguration:
+        BlockPublicAcls: True
+        BlockPublicPolicy: True
+        IgnorePublicAcls: True
+        RestrictPublicBuckets: True
+      BucketEncryption:
+        ServerSideEncryptionConfiguration:
+          - ServerSideEncryptionByDefault:
+              SSEAlgorithm: 'aws:kms'
+
+  CellSetsBucket:
+    Type: AWS::S3::Bucket
+    Properties:
+      BucketName: !Sub "cell-sets-${Environment}-${AWS::AccountId}"
+      PublicAccessBlockConfiguration:
+        BlockPublicAcls: True
+        BlockPublicPolicy: True
+        IgnorePublicAcls: True
+        RestrictPublicBuckets: True
+      BucketEncryption:
```

Comment:
do you know if we need to change anything in the way we are uploading or downloading data from these buckets because of the encryption?

Reply:
No, files are encrypted using AWS KMS. As I didnt specify an encryption key Amazon S3 automatically creates an AWS KMS key in the account the first time that you add an object encrypted with SSE-KMS to a bucket.

Reply:
Since every object is encrypted and decrypted manually with a different key, we might need to consider using S3 bucket keys, which will save money for requests to KMS because they will be for the entire bucket -https://docs.aws.amazon.com/AmazonS3/latest/userguide/bucket-key.html . What do you think?

Reply:
Yes, we should use that, good shout. Can you configure it?